File: todoapp.server/Models/List.cs
using System;
using System.Collections.Generic;

namespace todoapp.server.Models;

public partial class List
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public int? UserId { get; set; }

    public virtual User? User { get; set; }

    public virtual ICollection<Task> Tasks { get; set; } = new List<Task>();
}

File: todoapp.client/src/types/list.ts
import { ITask } from "./Task";
import { IUser } from "./User";

export interface IList {
    id: number;
    name: string;
    userId: number;
    user: IUser;
    tasks: ITask[];
}

export interface ListResponseDto {
    result: IList;
    numberOfTaskInfo: number;
}
